Libraries/src/Amazon.Lambda.Serialization.SystemTextJson/SourceGeneratorLambdaJsonSerializer.cs

```diff
@@ -95,6 +95,7 @@ public SourceGeneratorLambdaJsonSerializer(Action<JsonSerializerOptions> customi
         /// <param name="jsonSerializerContext"></param>
         /// <param name="customizer"></param>
         /// <param name="jsonWriterCustomizer"></param>
+        [Obsolete("The method is marked obsolete because the jsonSerializerContext parameter is unlikely to be created with the required JsonSerializerOptions for Lambda serialization. This will trigger confusing NullReferenceException.")]
```

Comment:
This overload was added last fall to pass in the instance of `JsonSerializerContext` to the `SourceGeneratorLambdaJsonSerializer`. It was done to avoid the reflection calls but the reflections used by `SourceGeneratorLambdaJsonSerializer` are annotated so they can be used in Native AOT. So this constructor wasn't strictly necessary. 

I'm marking as `Obsolete` because it turns out if you don't create the `JsonSerializerContext` with the correct json serialization options that `SourceGeneratorLambdaJsonSerializer` does like use case insensitive mapping the serialization fails at runtime with missing data. I lost a couple hours on this and I suspect others will too. To avoid this I think we should mark it as `Obsolete`.

Reply:
I'd add a resolution path for any users seeing this, maybe something like `You can now use the other constructors in a Native AOT deployment because the reflection calls have been annotated.` (assuming my understanding is correct)

Reply:
Done